sky/global_user_state.py

```diff
@@ -114,6 +128,24 @@ def add_or_update_cluster(cluster_name: str,
     cluster_launched_at = int(time.time()) if is_launch else None
     last_use = common_utils.get_pretty_entry_point() if is_launch else None
     status = ClusterStatus.UP if ready else ClusterStatus.INIT
+
+    cluster_hash = get_cluster_hash(cluster_name)
+
+    if not cluster_hash:
+        cluster_hash = str(uuid.uuid4())
+
+    requested_resources = get_requested_resources(cluster_handle)
+
+    metadata = get_cluster_history_metadata(cluster_hash)
+    if not metadata:
+        metadata = {}
+        metadata['usage_intervals'] = []
```

Comment:
Instead of having a dict, can we make it just a list and call it `usage_intervals`?

Reply:
I thought about this, but in the future it is possible we may have to add other keys to metadata. For that reason, I thought it was better to leave it is a dict since it will be more flexible.